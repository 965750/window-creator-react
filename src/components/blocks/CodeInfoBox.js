import React, { Component } from 'react'
import styles from './CodeInfoBox.module.scss'
import Cookies from 'js-cookie'

class CodeInfoBox extends Component {
    state = {
        showBox: false,
        visable: false,
    }

    componentDidMount() {
        if (!Cookies.get('showCodeInfoBox')) {
            setTimeout(() => {
                this.setState({
                    visable: true,
                })
            }, 2000)

            setTimeout(() => {
                this.setState({
                    showBox: true,
                })
            }, 1500)
        }
    }

    closeBox = () => {
        Cookies.set('showCodeInfoBox', '0', { expires: 2 })
        this.setState({
            showBox: false,
        })
    }

    render() {
        if (this.state.showBox) {
            return (
                <div
                    className={`${styles.box__wrapper} ${
                        this.state.visable ? styles['box__wrapper--active'] : ''
                    } z-30 p-2 pr-5 absolute right-20 bottom-20 border border-gullGray rounded bg-wildSand`}
                >
                    Are You interested in this app? If You want to see full code
                    checkout my Github repository here
                    <span
                        onClick={this.closeBox}
                        className={`${styles.box__close} absolute right-5 top-10 cursor-pointer`}
                    >
                        &#10005;
                    </span>
                </div>
            )
        } else {
            return null
        }
    }
}

export default CodeInfoBox
