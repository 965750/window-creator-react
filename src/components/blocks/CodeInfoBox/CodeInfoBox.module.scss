.box {

  &__wrapper {
    width: 200px;
    transition: all .5s;
    opacity: 0;

    &--active {
      opacity: 1;
    }
  }

  &__close {
    transform: translate(0, -50%)
  }
}