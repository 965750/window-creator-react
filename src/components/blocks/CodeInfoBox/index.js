import { FormattedMessage } from 'react-intl'

import Cookies from 'js-cookie'
import React, { Component } from 'react'
import styles from './CodeInfoBox.module.scss'

class CodeInfoBox extends Component {
  state = {
    showBox: false,
    visable: false,
  }

  componentDidMount() {
    if (!Cookies.get('showCodeInfoBox')) {
      setTimeout(() => {
        this.setState({
          visable: true,
        })
      }, 120000)

      setTimeout(() => {
        this.setState({
          showBox: true,
        })
      }, 119500)
    }
  }

  closeBox = () => {
    Cookies.set('showCodeInfoBox', '0', { expires: 2 })
    this.setState({
      showBox: false,
    })
  }

  render() {
    if (this.state.showBox) {
      return (
        <div
          className={`${styles.box__wrapper} ${
            this.state.visable ? styles['box__wrapper--active'] : ''
          } z-30 p-2 pr-5 fixed right-20 bottom-20 border border-gullGray rounded bg-wildSand`}
        >
          <p>
            <span>
              <FormattedMessage
                id="Are You interested in this app? If You want to see full code with tests checkout my "
                defaultMessage="Are You interested in this app? If You want to see full code with tests checkout my "
              />
            </span>
            <a href="https://github.com/965750/window-creator-react" className="hover:text-mountainMeadow text-malachite cursor-pointer">
              <FormattedMessage
                id="Github repository here"
                defaultMessage="Github repository here"
              />
            </a>
          </p>
          <button
            onClick={this.closeBox}
            className={`${styles.box__close} absolute right-5 top-10 cursor-pointer`}
          >
            &#10005;
          </button>
        </div>
      )
    }
    return null
  }
}

export default CodeInfoBox
