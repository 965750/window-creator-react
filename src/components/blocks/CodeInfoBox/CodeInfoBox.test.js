import '@testing-library/jest-dom/extend-expect'

import { applyMiddleware, compose, createStore } from 'redux';
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { render, cleanup, waitForElement } from '@testing-library/react'

import CodeInfoBox from './index'
import messages from '../../../messages-i18n'
import thunk from 'redux-thunk';
import React from 'react'

const initialStore = {
  theme: {
    lang: 'pl',
  },
}

const reducer = (state = initialStore, action) => {
  switch (action.type) {
    default:
      return state
  }
}

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunk),
  ),
)

const renderWithRedux = (component) => ({
  ...render(<Provider store={store}>{component}</Provider>),
})

describe('CodeInfoBox', () => {
  let wrapper
  const initialProps = {}

  const setUp = (props = initialProps) => {
    wrapper = renderWithRedux(
      <IntlProvider
        locale="en"
        messages={messages[store.getState().theme.lang]}
      >
        <CodeInfoBox {...props} />
      </IntlProvider>,
    )
  }

  afterEach(cleanup)

  it('should match snapshot', () => {
    setUp()

    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('should show up cookie banner after 2 minutes on page', async () => {
    jest.useFakeTimers()
    setUp()

    jest.runAllTimers()

    const cookieBox = await waitForElement(() => wrapper.getByText("repozytorium na Github'ie tutaj"))

    expect(cookieBox).toBeVisible()
  })
})
