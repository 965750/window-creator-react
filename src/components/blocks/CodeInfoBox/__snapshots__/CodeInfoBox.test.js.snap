// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeInfoBox should match snapshot 1`] = `<DocumentFragment />`;
